### PR TITLE
add latest docker config secret type

### PR DESCRIPTION
**Pull request by guangxuli**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36123)

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

File: docs/design/secrets.md [hunk 1]
```diff
@@ -324,6 +324,7 @@ const (
     SecretTypeOpaque              SecretType = "Opaque"                                 // Opaque (arbitrary data; default)
     SecretTypeServiceAccountToken SecretType = "kubernetes.io/service-account-token"    // Kubernetes auth token
     SecretTypeDockercfg           SecretType = "kubernetes.io/dockercfg"                // Docker registry auth
+    SecretTypeDockerConfigJson    SecretType = "kubernetes.io/dockerconfigjson"         // Latest Docker registry auth
```
Comment (liggitt) [inline docs/design/secrets.md]: where is this used?

Comment (guangxuli) [inline docs/design/secrets.md]: @liggitt 
definition at : https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L3267
used at: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L3144
https://github.com/kubernetes/kubernetes/blob/master/pkg/credentialprovider/keyring.go#L312
https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/runtime_conformance_test.go#L186
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation_test.go#L7913

**Comment by liggitt** [comment]
ah, missed that this was a doc update, LGTM

**Pull request merged**